app/controllers/favorites_controller.rb

```diff
@@ -9,14 +9,18 @@ def update
   end
 
   def index
-    @pets = Pet.find(session[:favorites])
+    @pets = Pet.find(favorites.contents)
   end
 
   def destroy
-    pet = Pet.find(params[:pet_id])
-    favorites.delete_pet(pet.id)
+    if params[:pet_id].nil?
```

Comment:
adds conditional to favorites#destroy - most RESTful approach I could come up with (but maybe still not RESTful at all...)

Reply:
if i had a better understanding about RESTful practices i could confirm or deny.